xbmc/cores/AudioEngine/Sinks/AESinkXAudio.cpp

```diff
@@ -887,3 +903,48 @@ bool CAESinkXAudio::IsUSBDevice()
 #endif
   return false;
 }
+
+bool CAESinkXAudio::AddEndOfStreamPacket()
+{
+  constexpr unsigned int frames{1};
+
+  XAUDIO2_BUFFER xbuffer = BuildXAudio2Buffer(nullptr, frames, 0);
```

Comment:
Are we really talking about frames here? Looks like one byte buffer? - Does not matter too much, but we should not interfer with what AE generates and gives us in frames vs. what we internaly do - as AE carefully tracks the frames we get, the payload they have and the playtime they have. As long as this only plays a role in Drain and internal members m_sinkFrames, etc. are back to zero after Drain has happened - all good.

Reply:
Yes there is. Though you need to distinguish between passthrough, where we use an existing packet and change its header to be pause burst (iec determines that from the header only) vs the usual PCM formats.

It is in the buffer and used for heating up and sync mechanisms.

Reply:
GenerateNoise is way overkill, and I just need a completely silent frame so I'm adding a GenerateSilence function to AEUtil. It might be useful for something else in the future.
In practice, the only formats that don't need all 0 bytes for silence are AV_SAMPLE_FMT_U8 and AV_SAMPLE_FMT_U8P